### PR TITLE
MdePkg: Remove duplicate source from BaseMemoryLib INF files

### PR DESCRIPTION
**Pull request by AshrafAliS**
# Description

There are some duplicate source files exits in the INF file, this patch is to remove those duplicate sources.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Basic Build

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Library/BaseMemoryLibOptPei/BaseMemoryLibOptPei.inf [hunk 1]
```diff
@@ -27,19 +27,21 @@
...
+  ScanMem8Wrapper.c
+  ZeroMemWrapper.c
+  CompareMemWrapper.c
+  SetMem64Wrapper.c
+  SetMem32Wrapper.c
+  SetMem16Wrapper.c
+  SetMemWrapper.c
+  CopyMemWrapper.c
+  IsZeroBufferWrapper.c
+  MemLibGuid.c
```
Comment (mdkinney) [inline MdePkg/Library/BaseMemoryLibOptPei/BaseMemoryLibOptPei.inf:42]: This part of the change looks like it is adding source files to the INF.  Not removing duplicates.  Please explain the reason for this change.

Comment (AshrafAliS) [reply MdePkg/Library/BaseMemoryLibOptPei/BaseMemoryLibOptPei.inf:42]: the same .C source were including in both IA32 and X64.
removed those .C source from IA32 and X64 and moved to common source section header.

**Review by mdkinney — APPROVED** [review]

**Comment by mdkinney** [comment]
This is not a critical bug fix, so i recommend not merging until after edk2-stable202408

**Review by niruiyu — APPROVED** [review]

**Comment by mdkinney** [comment]
@mergify rebase

**Pull request merged**